lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -174,41 +190,26 @@ def power_action_on_instance_by_uuid(self, uuid, power_action):
             Action.SHUTDOWN: self.api.shutdown_application_using_put,
             Action.STOPPED: self.api.stop_application_using_put,
             Action.RESTARTED: self.api.restart_application_using_put,
-            Action.FORCE_RESTARTED: self.api.force_restart_application_using_put,  # noqa
+            Action.FORCE_RESTARTED: self.api.force_restart_application_using_put,
             Action.PAUSED: self.api.pause_application_using_put,
             Action.ABSENT: self.api.delete_application_using_delete,
             Action.RESUMED: self.api.resume_application_using_put
         }
+        if power_action not in power_action_dict:
+            raise InvalidPowerActionException
         return power_action_dict[power_action](uuid)
 
 
-class ApplicationGroupResource(Resource):
-    resource_class = tacp.ApplicationGroupsApi
-
-    def filter(self, **filters):
-        return self.api.get_application_group_list_using_get(
-            **self.get_filters_kws(**filters)
-        )
-
-    @wait_to_complete
-    def create(self, name, datacenter_uuid):
-        return self.api.create_application_group_using_post(
-            tacp.ApiCreateApplicationGroupPayload(
-                name=name, datacenter_uuid=datacenter_uuid
-            )
-        )
-
-
 class VlanResource(Resource):
     resource_class = tacp.VlansApi
 
-    def filter(self, **filters):
-        return self.api.get_vlans_using_get(
-            **self.get_filters_kws(**filters)
-        )
+    filter_method = "get_vlans_using_get"
 
     def get_by_uuid(self, uuid):
-        return self.api.get_vlan_using_get(uuid)
+        resource = self.api.get_vlan_using_get(uuid)
+        if not resource:
+            raise UuidNotFoundException
```

Comment:
Not a blocker, but I think this is redundant here, returning `None` is just fine from my pov. It is also forcing outside callers to some kind of try except special treatment

Reply:
works for me, done